File: app/src/test/java/com/omar/retromp3recorder/app/share/TestSharingModule.java
package com.omar.retromp3recorder.app.share;

import javax.inject.Inject;
import javax.inject.Named;

import io.reactivex.Completable;
import io.reactivex.Observable;
import io.reactivex.subjects.Subject;

import static com.omar.retromp3recorder.app.di.MockModule.SHARING_SUBJECT;

public class TestSharingModule implements SharingModule {

    private final Subject<Event> events;

    @Inject
    TestSharingModule(@Named(SHARING_SUBJECT) Subject<Event> events) {
        this.events = events;
    }

    @Override
    public Completable share() {
        return Completable.complete();
    }

    @Override
    public Observable<Event> observeEvents() {
        return events;
    }
}

File: app/src/test/java/com/omar/retromp3recorder/app/share/TestSharingModule.kt
package com.omar.retromp3recorder.app.share

import com.omar.retromp3recorder.app.di.MockModule
import io.reactivex.Completable
import io.reactivex.Observable
import io.reactivex.subjects.Subject
import javax.inject.Inject
import javax.inject.Named

open class TestSharingModule @Inject internal constructor(
    @param:Named(MockModule.SHARING_SUBJECT) private val events: Subject<SharingModule.Event>
) :
    SharingModule {
    override fun share(): Completable {
        return Completable.complete()
    }

    override fun observeEvents(): Observable<SharingModule.Event> {
        return events
    }
}
